src/BotwinExtensions.cs

```diff
@@ -164,13 +172,14 @@ public static void AddBotwin(this IServiceCollection services, params Assembly[]
             var modules = assemblies.SelectMany(x => x.GetTypes().Where(t => typeof(BotwinModule).IsAssignableFrom(t) && t != typeof(BotwinModule)));
             foreach (var module in modules)
             {
-                services.AddTransient(typeof(BotwinModule), module);
+                services.AddScoped(module);
```

Comment:
Why not leave it as transient?

Reply:
@khellang why is the module added twice?

Reply:
@jchannon You need to add the module once as `BotwinModule`, so you can resolve all modules by requesting `IEnumerable<BotwinModule>` from DI. And then, you need to register each module as itself, so you can get the exact module later when you retrieve the module in the path handler.

Reply:
Seems odd 😄

Reply:
Maybe, but basically you are just assigning each module two “labels” you can refer it to. Once as the module itself, and once with it being *a* module.